kotlin/workflow-ui-android/src/main/java/com/squareup/workflow/ui/WorkflowViewStub.kt

```diff
@@ -0,0 +1,62 @@
+package com.squareup.workflow.ui
+
+import android.content.Context
+import android.util.AttributeSet
+import android.view.View
+import android.view.ViewGroup
+
+/**
+ * A placeholder [View] that can replace itself with ones driven by workflow renderings,
+ * similar to [android.view.ViewStub].
+ */
+class WorkflowViewStub(
+  context: Context,
+  attributeSet: AttributeSet? = null
+) : View(context, attributeSet) {
+  init {
+    setWillNotDraw(true)
+  }
+
+  /**
+   * By starting as our own delegate, we make the parent checking code in [update] simpler.
+   */
+  private var _actual: View = this
+
+  /**
+   * On-demand access to the delegate established by the last call to [update],
+   * or null if none has been set yet.
+   */
+  val actual: View? get() = _actual.takeIf { it !is WorkflowViewStub }
+
+  /**
+   * Replaces this view with one that can display [rendering]. If it
+   * has already been replaced, updates that replacement if it [canShowRendering].
+   * If it can't, a new replacement is put in place.
+   *
+   * @return the view that showed [rendering]
+   * @throws RuntimeException if [rendering] cannot be displayed, typically indicating
+   * a misconfigured [registry]
+   */
+  fun update(
+    rendering: Any,
+    registry: ViewRegistry
+  ): View {
+    _actual.takeIf { it.canShowRendering(rendering) }
+        ?.let {
```

Comment:
I liked this better as an vanilla `if` statement, I don't think the fancy scoping functions add any value.

I'm curious if the compiler or R8 or something can optimize this down to a single `if`, instead of a local nullable variable with a null check. But I think the `if` is just more readable regardless.

Reply:
I strongly prefer the mapping functions when dealing with mutable fields, even when smart casting isn't particularly needed. And I find this very readable.

Reply:
I really hate this, but it's correct so I can't stop you.